src/test/scala/is/hail/nativecode/NativeCodeSuite.scala

```diff
@@ -50,4 +50,82 @@ class NativeCodeSuite extends SparkSuite {
     assert(a.use_count() == 1)
   }
 
+  @Test def testNativeGlobal() = {
+    var ret: Long = -1
+    val st = new NativeStatus()
+    val globalModule = new NativeModule("global")
+    val funcHash1 = globalModule.findLongFuncL1(st, "hailTestHash1")
+    if (st.fail) error(s"${st}")
+    assert(st.ok)
+    val funcHash8 = globalModule.findLongFuncL8(st, "hailTestHash8")
+    if (st.fail) error(s"${st}")
+    assert(st.ok)
+    ret = funcHash8(st, 0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x8)
+    assert(ret == 0x87654321L)
+    val t0 = System.currentTimeMillis()
+    var sum: Long = 0
+    val numCalls = 100*1000000
+    var countdown = numCalls
+    while (countdown > 0) {
+      sum = funcHash1(st, sum)
+      countdown -= 1
+    }
+    val t1 = System.currentTimeMillis()
+    val usecsPerCall = ((t1 - t0) * 1000.0) / numCalls
+    info(s"funcHash1() ~ ${usecsPerCall}usecs")
+    assert(usecsPerCall < 0.2)
+  }
+
+  @Test def testNativeBuild() = {
+    val sb = new StringBuilder()
+    sb.append("#include \"hail/hail.h\"\n")
+    sb.append("NAMESPACE_HAIL_MODULE_BEGIN\n")
+    sb.append("\n")
+    // A very simple function
+    sb.append("long testFunc1(NativeStatus* st, long a0) { return a0+1; }\n\n")
+    // Now declare our own NativeObj
+    sb.append("class MyObj : public NativeObj {\n")
+    sb.append("public:\n")
+    sb.append("  int val_;\n")
+    sb.append("\n")
+    sb.append("  MyObj(int val) : val_(val) { }\n")
+    sb.append("  ~MyObj() { }\n")
+    sb.append("  const char* get_class_name() { return \"MyObj\"; }\n")
+    sb.append("};\n")
+    sb.append("\n")
+    sb.append("NativeObjPtr makeMyObj(NativeStatus*, long val) {\n")
+    sb.append("  return std::make_shared<MyObj>(val);\n")
+    sb.append("}\n")
+    sb.append("\n")
+    sb.append("NAMESPACE_HAIL_MODULE_END\n")
+    val options = "-ggdb -O2"
+    val st = new NativeStatus()
+    val mod = new NativeModule(options, sb.toString(), true)
+    mod.findOrBuild(st)
+    if (st.fail) error(s"${st}")
+    assert(st.ok)
+    val testFunc1 = mod.findLongFuncL1(st, "testFunc1")
+    if (st.fail) error(s"${st}")
+    assert(st.ok)
+    val ret = testFunc1(st, 6)
+    info(s"testFunc(6) returns ${ret}")
```

Comment:
generally, we avoid logging in tests. For this I'd just write:

```scala
assert(testFunc1(st, 6) == 7)
```
This `assert` is actually a macro that will print a nice message if this isn't true.